src/manifest.c

```diff
@@ -594,8 +617,15 @@ struct manifest *load_mom(int version)
 
 	string_or_die(&filename, "%s/%i/Manifest.MoM", state_dir, version);
 	string_or_die(&url, "%s/%i/Manifest.MoM", content_url, version);
+verify_mom:
 	if (!download_and_verify_signature(url, filename)) {
 		if (sigcheck) {
+			/* cleanup and try one more time, statedir could have got corrupt/stale */
+			if (retried == false) {
+				remove_manifest_files("MoM", version);
+				retried = true;
+				goto verify_mom;
+			}
```

Comment:
Okay, I see that you're right about the sigcheck == false path; signature check failures are not enforced, so any number of additional errors might appear and not worth adding special handling.

I agree with moving `manifest_from_file` before the signature check.